node/cn/backend_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package cn
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/datasync/downloader"
+	"github.com/klaytn/klaytn/node/cn/mocks"
+	"github.com/klaytn/klaytn/params"
+	mocks2 "github.com/klaytn/klaytn/work/mocks"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func newCN(t *testing.T) (*gomock.Controller, *MockBackendProtocolManager, *mocks.MockMiner, *CN) {
+	mockCtrl := gomock.NewController(t)
+	mockProtocolManager := NewMockBackendProtocolManager(mockCtrl)
+	mockMiner := mocks.NewMockMiner(mockCtrl)
+	return mockCtrl, mockProtocolManager, mockMiner,
+		&CN{protocolManager: mockProtocolManager, miner: mockMiner}
+}
+
+func TestCN_AddLesServer(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	cn := &CN{}
+	les := mocks.NewMockLesServer(mockCtrl)
+	les.EXPECT().SetBloomBitsIndexer(nil).Times(1)
+	cn.AddLesServer(les)
+}
+
+func TestCN_CheckSyncMode(t *testing.T) {
+	c := &Config{SyncMode: downloader.FastSync}
+	assert.NoError(t, checkSyncMode(c))
+
+	c.SyncMode = downloader.FullSync
+	assert.NoError(t, checkSyncMode(c))
+
+	c.SyncMode = downloader.LightSync
+	assert.Equal(t, errCNLightSync, checkSyncMode(c))
+}
+
+func TestCN_SetEngineType(t *testing.T) {
+	cc := &params.ChainConfig{}
+	originalEngineType := types.EngineType
+
+	setEngineType(cc)
+	assert.Equal(t, originalEngineType, types.EngineType)
+
+	cc.Clique = &params.CliqueConfig{}
+	setEngineType(cc)
+	assert.Equal(t, types.Engine_Clique, types.EngineType)
+
+	cc.Istanbul = &params.IstanbulConfig{}
+	setEngineType(cc)
+	assert.Equal(t, types.Engine_IBFT, types.EngineType)
+}
+
+func TestCN_SetAcceptTxs(t *testing.T) {
+	{
```

Comment:
```suggestion
```

Reply:
@melvin Will you add another test case in next PR for covering below code?
```
// setAcceptTxs sets AcceptTxs flag in 1CN case to receive tx propagation.
func (s *CN) setAcceptTxs() error {
	if s.chainConfig.Istanbul != nil {
		istanbulExtra, err := types.ExtractIstanbulExtra(s.blockchain.Genesis().Header())
		if err != nil {
			return err
		} else {
			if len(istanbulExtra.Validators) == 1 {
				atomic.StoreUint32(&s.protocolManager.acceptTxs, 1)
			}
		}
	}
	return nil
}```

Reply:
@KimKyungup Yes, I will. Currently this test covers only a part of the function. But properly setting `ExtractIstanbulExtra` is needed to extend the coverage. I will update the test in another PR. cc @jeonghonah